onboarding/src/Components/FormsEdit/FormsEditPage.js

```diff
@@ -38,38 +40,87 @@ function FormsEditPage() {
 
   useEffect(() => {
     setSections(sortedSections);
+    // Shallow copy an Array of Objects
+    setSectionsCopy(JSON.parse(JSON.stringify(sortedSections)));
     updateMaxOrder(sortedSections?.length);
     setLoading(isLoading);
     setErrorMessage(error);
   }, [sortedSections, isLoading, error]);
 
+  const showAutosaveInfo = () => {
+    // Show info "Zapisano zmiany" for 3 sec. when the changes were saved
+    if(saveOnDemand !== true) {
+      setIsAutosave(true);
+      const timer = setTimeout(() => {
+          setIsAutosave(false);
+      }, 3000);
+
+      return () => {
+          clearTimeout(timer);
+      }
+    };
+  }
+  useEffect(() => {
+    // Compare two Arrays of Objects
+    if(sections && sectionsCopy && JSON.stringify(Object.values(sections)) == JSON.stringify(Object.values(sectionsCopy))) return;
+    if(sections && !sections.some(section => section.title === "") && saveOnDemand !== true) {
+      setErrorMessage("");
+
+      // Save changes after 10 sec. form last change
+      const saveTimeout = setTimeout(
+        () => {
+          if(saveOnDemand !== true) {
+            FormSectionsAPI.saveAll(sections)
+            // FormSectionsAPI.saveAll(sections, updateUnsetAsNew)
+              .catch((error) => setErrorMessage(error.message))
+              .then((result) => {
+                // Shallow copy an Array of Objects
+                setSectionsCopy(JSON.parse(JSON.stringify(result)));
+                setSections(result);
+              })
+              .finally(() => {
+                showAutosaveInfo();
+              });
+            }
+          },
+          10000
+          );
+      return () => clearTimeout(saveTimeout);
+    } 
+  }, [sections]);
+
   const hideModal = () => {
     setShowSaveModal(false);
+    setSaveOnDemand(false);
   };
-
+  
   const updateUnsetAsNew = function(newSections){
     if(typeof newSections === 'undefined' || newSections === null)
-      return;
-
-    let newSections2 = newSections.map( (section) => {
-      if(section.hasOwnProperty('isNew') )
-        section.isNew = false;
```

Comment:
It was (still is) necessary to set the section not being new when the `POST` request was done as it is not new on the server anymore. By removing/commenting this you make the request being constantly `POST` for already created sections so whenever yous clicks "Zapisz pytania" the server will duplicate them. We are working on this duplication and it is issue #180 .
So in short: it makes the request being `POST` only once and only for sections the user created by click of corresponding button on the right panel.

Reply:
Ok, I didn't notice duplication on my computer. I will redo this change.